File: leetcode/C++/RemoveDuplicatesfromSortedList.cpp
// Given a sorted linked list, delete all duplicates such that each element appear only once.

// For example,
// Given 1->1->2, return 1->2.
// Given 1->1->2->3->3, return 1->2->3.


/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *deleteDuplicates(ListNode *head) {
        if(head == NULL || head->next == NULL)
            return head; 
        ListNode *p;
        ListNode *q;
        q=head;
        p=head->next; 
        while(p!=NULL)
        {
            if(p->val==q->val)
            {
                q->next=p->next;
                p=q->next;
            }else{
           		p=p->next;
            	q=q->next;
            }    
        }    
        return head;  
    }
};

File: leetcode/leetcode_Easy/RemoveDuplicatesfromSortedList.java
package leetcode_Easy;

public class RemoveDuplicatesfromSortedList {
	public ListNode deleteDuplicates(ListNode head) {
        if(head==null || head.next==null) return head;
        ListNode p=head;
        
        while(p!=null && p.next!=null){
        	if(p.val==p.next.val){
        		p.next=p.next.next;
        	}else{
        		p=p.next;
        	}
        }
        
        return head;
        
    }
	
	public ListNode deleteDuplicates2(ListNode head) {
        if(head == null)
            return null;
        if(head.next != null && head.next.val == head.val) {
            head.next = head.next.next;
            head = deleteDuplicates(head);
        }
        else
            head.next = deleteDuplicates(head.next);

        return head;
    } 
}
